File: 2-concurrency-and-paralelism/python/src/primes.py
"""
Funções compartilhadas para o benchmark de concorrência/paralelismo em Python.

Inclui:
* `is_prime(n)`: verificação simples de primalidade.
* `generate_data(n)`: gera `n` números inteiros determinísticos usando um LCG.
"""
from __future__ import annotations

import math


def is_prime(n: int) -> bool:
    """Retorna True se `n` for primo, False caso contrário."""
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    limit = int(math.isqrt(n))
    for i in range(3, limit + 1, 2):
        if n % i == 0:
            return False
    return True


def generate_data(count: int) -> list[int]:
    """Gera uma lista determinística de inteiros usando LCG."""
    data: list[int] = [0] * count
    seed = 42
    for i in range(count):
        seed = (seed * 1664525 + 1013904223) & 0xFFFFFFFF
        data[i] = 100_000 + (seed % 900_000)
    return data

File: encontros/2025/2025_12_10__19_00/go/src/primes/primes.go
package primes

func IsPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	for i := 3; i*i <= n; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

// GenerateData gera um slice de inteiros determinísticos usando LCG.
func GenerateData(count int) []int {
	data := make([]int, count)
	var seed uint32 = 42
	for i := 0; i < count; i++ {
		seed = seed*1664525 + 1013904223
		data[i] = 100000 + int(seed%900000)
	}
	return data
}
